commands/checkout.go

```diff
@@ -119,9 +119,19 @@ func transformCheckoutArgs(args *Args) error {
 				newBranchName = fmt.Sprintf("%s-%s", pullRequest.Head.Repo.Owner.Login, pullRequest.Head.Ref)
 			}
 		}
-		refSpec := fmt.Sprintf("refs/pull/%s/head:%s", id, newBranchName)
 		newArgs = append(newArgs, newBranchName)
-		args.Before("git", "fetch", baseRemote.Name, refSpec)
+
+		ref := fmt.Sprintf("refs/pull/%s/head", id)
+		args.Before("git", "fetch", baseRemote.Name, fmt.Sprintf("%s:%s", ref, newBranchName))
+
+		remote := baseRemote.Name
+		mergeRef := ref
+		if pullRequest.MaintainerCanModify && pullRequest.Head.Repo != nil {
+			remote = pullRequest.Head.Repo.SshUrl
```

Comment:
People who have `hub.protocol` set to "https" will want that this URL is `https:`, because they might not have SSH keys configured for GitHub, and instead use HTTPS authentication with credential helper.

Perhaps you could [convert a Repo object to Project](https://github.com/github/hub/blob/fe5b59b5489f2ac647c51914bf8e1b0e9993451f/commands/cherry_pick.go#L86) and [use its `GitURL()` method](https://github.com/github/hub/blob/fe5b59b5489f2ac647c51914bf8e1b0e9993451f/github/project.go#L75-L78)? Then you don't have to worry about the logic of constructing a proper clone/push URL, and you don't have to add the extra `SshUrl` property to the struct either.